src/Symfony/Component/Form/Form.php

```diff
@@ -502,10 +502,27 @@ public function submit($submittedData, $clearMissing = true)
         // they are collectable during submission only
         $this->errors = [];
 
+        $dispatcher = $this->config->getEventDispatcher();
+
         // Obviously, a disabled form should not change its data upon submission.
         if ($this->isDisabled()) {
             $this->submitted = true;
 
+            if ($dispatcher->hasListeners(FormEvents::PRE_SUBMIT)) {
+                $event = new FormEvent($this, $submittedData);
```

Comment:
@fancyweb Ignoring the submitted data is expected to me because one would not expect to have data changed for a fully disabled form. Though your comment made me aware of the fact that we probably need to pass `$this->getViewData()` instead of the submitted data here, right? This *should* not have any side effects as we never deal with any potential changes made by the event listeners.

Reply:
No, I think we should still pass the submitted data, but probably the final version of it (cf the transformation done on it starting line 541), to handle the same data in the listener whether the form is disabled or not.

My whole point is about backward compatibility. If I have a form that can be disabled and that listen on those 3 events, those listeners are now going to be executed but they are probably not ready for it, because the data they are going to receive is not going to be the expected one.

Reply:
Maybe this logic could be executed only for the root form (validation is by cascade only). What about adding a comment about why we dispatch here (i.e "Allow validation of initial state in case fields are dependent, or all disabled.")